src/tr1/game/gamebuf.h

```diff
@@ -9,45 +11,6 @@
 // intentionally not implemented. To use more dynamic memory management, use
 // Memory_Alloc / Memory_Free.
 
-typedef enum {
-    GBUF_TEXTURE_PAGES,
-    GBUF_MESH_POINTERS,
-    GBUF_MESHES,
-    GBUF_ANIMS,
-    GBUF_ANIM_CHANGES,
-    GBUF_ANIM_RANGES,
-    GBUF_ANIM_COMMANDS,
-    GBUF_ANIM_BONES,
-    GBUF_ANIM_FRAMES,
-    GBUF_ROOMS,
-    GBUF_ROOM_MESH,
-    GBUF_ROOM_DOOR,
-    GBUF_ROOM_SECTOR,
-    GBUF_ROOM_LIGHTS,
-    GBUF_ROOM_STATIC_MESHES,
-    GBUF_FLOOR_DATA,
-    GBUF_ITEMS,
-    GBUF_CAMERAS,
-    GBUF_SOUND_FX,
-    GBUF_BOXES,
-    GBUF_OVERLAPS,
-    GBUF_GROUNDZONE,
-    GBUF_FLYZONE,
-    GBUF_ANIMATING_TEXTURE_RANGES,
-    GBUF_CINEMATIC_FRAMES,
-    GBUF_LOADDEMO_BUFFER,
-    GBUF_EXTRA_DOOR_STUFF,
-    GBUF_EFFECTS,
-    GBUF_CREATURE_INFO,
-    GBUF_CREATURE_LOT,
-    GBUF_SAMPLE_INFOS,
-    GBUF_SAMPLES,
-    GBUF_TRAP_DATA,
-    GBUF_CREATURE_DATA,
-    GBUF_VERTEX_BUFFER,
-} GAME_BUFFER;
-
 void GameBuf_Init(void);
 void GameBuf_Reset(void);
-void *GameBuf_Alloc(int32_t alloc_size, GAME_BUFFER buffer);
```

Comment:
No need to repeat the function declarations that we already declare in libtrx.

Reply:
`GameBuf_Init` has a different signature in the two games, and TR2 has a `GameBuf_Free` while TR1 doesn't. I figured just to move the one function we need for now, so the others are only declared in the corresponding game header.

TR2 still refers to a global buffer variable too. Quite happy to make everything the same though in this PR, but maybe better as a separate task?

Reply:
Good call – no, let's tackle that separately.